### PR TITLE
Fix MM-57090

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Switching the authentication type to Email, the server needs to check if
1. Email as an authentication mechanism is enabled, i.e. if `EnableSignUpWithEmail` is `true` and
2. The user can log in after making the authentication switch, i.e., if `EnableSignInWithEmail` or `EnableSignInWithUsername` is enabled.

The PR also updates the webapp to reflect these checks.

See https://hub.mattermost.com/private-core/pl/8xmom5f53bnkpb9axr4h8j113w for more details.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57091

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by sbishel** [comment]
@hanzei Please summarize the changes in the description.

**Review by esarafianou — APPROVED** [review]

**Comment by hanzei** [comment]
@sbishel Thanks for the suggestion. I've updated the PR description.

**Review by sbishel — APPROVED** [review]
LGTM

File: webapp/channels/src/components/user_settings/security/index.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ function mapStateToProps(state: GlobalState, ownProps: Props) {
     const userHasTokenRole = UserUtils.hasUserAccessTokenRole(ownProps.user.roles) || UserUtils.isSystemAdmin(ownProps.user.roles);
 
     const enableOAuthServiceProvider = config.EnableOAuthServiceProvider === 'true';
-    const enableSignUpWithEmail = config.EnableSignUpWithEmail === 'true';
+    const allowedToSwitchToEmail = config.EnableSignInWithEmail === 'true' && (config.EnableSignInWithEmail === 'true' || config.EnableSignInWithUsername === 'true');
```
Comment (BenCookie95) [inline webapp/channels/src/components/user_settings/security/index.ts:38]: Should `config.EnableSignInWithEmail === 'true' && ....` be `config.EnableSignUpWithEmail === 'true' &&....`?

**Review by BenCookie95 — CHANGES_REQUESTED** [review]
Nice, just one small issue

File: webapp/channels/src/components/user_settings/security/index.ts (hunk 1 above)
Comment (hanzei) [reply webapp/channels/src/components/user_settings/security/index.ts]: Excellent catch, thanks @BenCookie95 :100:

**Review by BenCookie95 — APPROVED** [review]

**Pull request merged**